# Energy

In [1]:
using JLD, FileIO

In [82]:
f = File(format"JLD","../data/uci/energy")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (768, 8)
Output dimension: (768, 2)


In [83]:
files = map(i -> "energy_$(i).csv", 1:5);

In [84]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 10.1809   3.1203   3.119    2.02348  2.3187 
  9.55063  3.05243  3.05173  2.11569  2.57756
  9.94643  3.41418  3.41419  2.15133  2.30241
  9.38786  3.33076  3.33045  2.03239  2.03221
  9.78676  3.3593   3.35922  1.94144  1.94145

header:
* mean
* LLS
* Ridge
* GP
* SPN-GP 

In [85]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.2f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $9.77 \pm 0.14$
LLS & $3.26 \pm 0.07$
Ridge & $3.25 \pm 0.07$
GP & $2.05 \pm 0.04$
SPN-GP & $2.23 \pm 0.11$


Mean & $9.77 \pm 0.14$
LLS & $3.26 \pm 0.07$
Ridge & $3.25 \pm 0.07$
GP & $2.05 \pm 0.04$
SPN-GP & $2.07 \pm 0.04$

# Concrete

In [90]:
f = File(format"JLD","../data/uci/concrete")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (1030, 8)
Output dimension: (1030, 1)


In [91]:
files = map(i -> "concrete_$(i).csv", 1:5);

In [92]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 16.4522   9.55099   9.55226  6.17692  6.17644
 16.6221  10.7554   10.7566   6.38719  6.38748
 16.7283  10.386    10.3835   6.40419  6.39088
 16.2798  10.9258   10.9256   6.54563  6.55544
 16.1821  10.0504   10.0506   5.73197  5.72963

In [94]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.3f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $16.453 \pm 0.10$
LLS & $10.334 \pm 0.25$
Ridge & $10.334 \pm 0.25$
GP & $6.249 \pm 0.14$
SPN-GP & $6.248 \pm 0.14$


Mean & $16.45 \pm 0.10$
LLS & $10.33 \pm 0.25$
Ridge & $10.33 \pm 0.25$
GP & $6.25 \pm 0.14$
SPN-GP & $6.27 \pm 0.20$

# CCPP

In [2]:
f = File(format"JLD","../data/uci/ccpp")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (9568, 4)
Output dimension: (9568, 1)


In [3]:
files = map(i -> "ccpp_$(i).csv", 1:5);

In [4]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 16.8617  4.73219  4.73219  4.22023  4.21914
 16.8356  4.61411  4.61412  4.08975  4.08959
 16.9853  4.47446  4.47448  3.9564   3.95601
 16.996   4.48978  4.48976  3.91779  3.91617
 16.8035  4.55037  4.55035  4.01961  4.01944

In [5]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.2f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $16.90 \pm 0.04$
LLS & $4.57 \pm 0.05$
Ridge & $4.57 \pm 0.05$
GP & $4.04 \pm 0.05$
SPN-GP & $4.04 \pm 0.05$


In [118]:
10^2

100